apm-sniffer/apm-sdk-plugin/mongodb-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v4/support/MongoInfoHelper.java

```diff
@@ -140,4 +146,65 @@ private static String limitFilter(String filter) {
         }
     }
 
+    /**
+     * Get the databaseName
+     *
+     * @param obj operation
+     * @return databaseName
+     */
+    @SuppressWarnings("rawtypes")
+    public static String getMongoDatabaseName(Object obj) throws NoSuchFieldException, IllegalAccessException {
+        String databaseName = "";
+        if (obj instanceof CountOperation) {
+            Field namespace = CountOperation.class.getDeclaredField(NAMESPACE_FIELD_NAME);
+            namespace.setAccessible(true);
+            databaseName = getOrDefaultDatabaseName((MongoNamespace) namespace.get(obj));
```

Comment:
You are doing `instance of` and reflection for every request multiple times, usually this would cost significant resources of CPU and cause extra latency.

Reply:
This will slow down the extra time of tracing from 10 nanoseconds to 100+ ns or more, determined by how many `instanceof`s you have to run before the reflection.

As you are only trying to read DB instance info, can't you get that from the constructor of those `Operation`s?  You could instrument those and use dynamic field to carry all necessary information. 
I believe from there, you don't need `instanceof` anymore.

Reply:
> This will slow down the extra time of tracing from 10 nanoseconds to 100+ ns or more, determined by how many `instanceof`s you have to run before the reflection.
> 
> As you are only trying to read DB instance info, can't you get that from the constructor of those `Operation`s? You could instrument those and use dynamic field to carry all necessary information. I believe from there, you don't need `instanceof` anymore.

Okay, thank you for your prompt. I will modify this part of the code